File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/domain/java/Customer.java
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.domain.java;

// Customer is also modeled as a record.
// Java records automatically generate constructor,
// accessors, equals(), hashCode(), and toString().
public record Customer(
        String id,
        String fullName
) {

    public Customer {
        if (id == null || id.isBlank()) {
            throw new IllegalArgumentException("Customer id cannot be empty");
        }

        if (fullName == null || fullName.isBlank()) {
            throw new IllegalArgumentException("Customer name cannot be empty");
        }
    }
}

File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/domain/kotlin/Customer.kt
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.domain.kotlin

// Customer is compact and expressive.
// The domain model is easy to read because properties are declared directly.
data class Customer(
    val id: String,
    val fullName: String
) {
    init {
        require(id.isNotBlank()) {
            "Customer id cannot be empty"
        }

        require(fullName.isNotBlank()) {
            "Customer name cannot be empty"
        }
    }
}